test/boost/big_decimal_test.cc

```diff
@@ -203,3 +203,112 @@ BOOST_AUTO_TEST_CASE(test_big_decimal_sub) {
+    const auto fmt_ordering = [] (std::strong_ordering r) {
+        if (r == std::strong_ordering::less) {
+            return "<";
+        } else if (r == std::strong_ordering::equal) {
+            return "=";
+        } else if (r == std::strong_ordering::greater) {
+            return ">";
+        } else {
+            return "?";
+        }
+    };
+    // Some of the numbers in this test are truly humongous, converting them to string will stress the machine and will make the test output unusable.
+    // So avoid rendering the numbers using big_decimal::to_string(), just print the unscaled value and the scale instead.
+    const auto fmt_decimal = [] (const big_decimal& x) {
+        return fmt::format("{}**{}", x.unscaled_value().str(), -int64_t(x.scale()));
+    };
+    auto test_cmp = [&] (const big_decimal& a, const big_decimal& b, std::strong_ordering expected_result, std::source_location sl = std::source_location::current()) {
+        fmt::print("checking {} {} {} @ {}:{}\n", fmt_decimal(a), fmt_ordering(expected_result), fmt_decimal(b), sl.file_name(), sl.line());
+        auto res = a <=> b;
+        if (res != expected_result) {
+            BOOST_FAIL(fmt::format("{} <=> {}: expected result: {}, but got {}", fmt_decimal(a), fmt_decimal(b), fmt_ordering(expected_result), fmt_ordering(res)));
+        }
+    };
+
+    // Test compare with same scale, should hit early return for same scale.
+    test_cmp(big_decimal("1"), big_decimal("-1"), std::strong_ordering::greater);
+    test_cmp(big_decimal("1.3345E3"), big_decimal("-1.3344E3"), std::strong_ordering::greater);
+    test_cmp(big_decimal("18E2"), big_decimal("2E2"), std::strong_ordering::greater);
+
+    // Test compare with different signs, should hit early return for different signs.
+    test_cmp(big_decimal("-1.3345E3"), big_decimal("1"), std::strong_ordering::less);
+    test_cmp(big_decimal("1.3345E3"), big_decimal("-1.3344E5"), std::strong_ordering::greater);
+
+    // Test compare against 0, should hit early return for different zero.
+    test_cmp(big_decimal("0"), big_decimal("0"), std::strong_ordering::equal);
+    test_cmp(big_decimal("0E6"), big_decimal("0E1"), std::strong_ordering::equal);
+    test_cmp(big_decimal("1"), big_decimal("0"), std::strong_ordering::greater);
+    test_cmp(big_decimal("1"), big_decimal("0E7"), std::strong_ordering::greater);
+    test_cmp(big_decimal("1.32432345E7"), big_decimal("0"), std::strong_ordering::greater);
+    test_cmp(big_decimal("-1"), big_decimal("0"), std::strong_ordering::less);
+    test_cmp(big_decimal("-1"), big_decimal("0E-4"), std::strong_ordering::less);
+    test_cmp(big_decimal("-1.3345E3"), big_decimal("0"), std::strong_ordering::less);
+    test_cmp(big_decimal("0"), big_decimal("1"), std::strong_ordering::less);
+    test_cmp(big_decimal("0"), big_decimal("187687E3"), std::strong_ordering::less);
+    test_cmp(big_decimal("0"), big_decimal("187687E3"), std::strong_ordering::less);
```

Comment:
The two last checks are identical, I assume you wanted to do something different?

Reply:
I think I wanted to have a test with a negative number too. I will adjust.

Reply:
Made the last one test with negatives in v5.